Applications/_Plugins/Common/MipsToQImage.h

```diff
@@ -0,0 +1,33 @@
+// AMD AMDUtils code
+// 
+// Copyright(c) 2017 Advanced Micro Devices, Inc.All rights reserved.
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#ifndef CMP_MIPS_TO_QIMAGE_H_INCLUDE
+#define CMP_MIPS_TO_QIMAGE_H_INCLUDE
+
+#include <Compressonator.h>
+#include <Common.h>
+
+// forward declaration
+class QImage;
+
+int QImage2MIPS(QImage *qimage, CMIPS *m_CMips, MipSet *pMipSet);
```

Comment:
This is the extraction of code that was `extern`ed in a couple places into one library to prevent linker errors and make it easier to track and maintain.